platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/capabilities/CapabilitiesLocalComponentIntegrationTest.groovy

```diff
@@ -76,5 +77,64 @@ class CapabilitiesLocalComponentIntegrationTest extends AbstractIntegrationSpec
         then:
         outputContains("Could not resolve project :.")
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/26377")
+    def "ResolvedVariantResults reported by ResolutionResult and ArtifactCollection have same capabilities when they are added to configuration in hierarchy"() {
+        settingsFile << "include 'producer'"
+        file("producer/build.gradle") << """
+            group="com.foo"
+
+            task zip(type: Zip) {
+                archiveFileName = "producer.zip"
+                destinationDirectory = layout.buildDirectory
+            }
+
+            configurations {
+                dependencyScope("api") {
+                    outgoing {
```

Comment:
In general I find it odd that we inherit capabilities from parent configurations in the hierarchy. Especially since you need to go through `outgoing` to add capabilities to a configuration. 

Should we deprecate this?

Reply:
This seems weird to me too.

A capability is purely a consumable/variant concern.  It has no meaning on resolvable configurations or dependency scopes.

This falls into the same bucket as using a dependency scope to pass around artifacts.   

I think we would deprecate this when we go through all of the `Configuration` methods/behaviors and decide which ones belong to which roles.